src/signature.rs

```diff
@@ -56,179 +46,27 @@ pub struct Signature {
     ///
     /// This digest is then interpreted as a `Scalar` and reduced into an
     /// element in ℤ/lℤ.
-    pub(crate) s: Scalar,
-}
-
-impl Clone for Signature {
-    fn clone(&self) -> Self {
-        *self
-    }
+    fn s(&self) -> Scalar;
 }
 
-impl Debug for Signature {
-    fn fmt(&self, f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
-        write!(f, "Signature( R: {:?}, s: {:?} )", &self.R, &self.s)
-    }
-}
-
-#[cfg(feature = "legacy_compatibility")]
-#[inline(always)]
-fn check_scalar(bytes: [u8; 32]) -> Result<Scalar, SignatureError> {
-    // The highest 3 bits must not be set.  No other checking for the
-    // remaining 2^253 - 2^252 + 27742317777372353535851937790883648493
-    // potential non-reduced scalars is performed.
-    //
-    // This is compatible with ed25519-donna and libsodium when
-    // -DED25519_COMPAT is NOT specified.
-    if bytes[31] & 224 != 0 {
-        return Err(SignatureError(InternalError::ScalarFormatError));
-    }
-
-    Ok(Scalar::from_bits(bytes))
-}
-
-#[cfg(not(feature = "legacy_compatibility"))]
-#[inline(always)]
-fn check_scalar(bytes: [u8; 32]) -> Result<Scalar, SignatureError> {
```

Comment:
This was unfortunate to lose in the rebase, and perhaps there's something we can still do yet to retain it, but note that the `check_scalar` check above (for the highest 3 bits being unset) was adapted into the `ed25519` crate:

https://github.com/RustCrypto/signatures/blob/master/ed25519/src/lib.rs#L104